### ISSUE TITLE
Deep Bayesian Model Discovery without using NeuralODE object

### ISSUE THREAD

**Issue by gurtajbir** [SciML/SciMLDocs#169]
Hi everyone. I am trying to implement the [Deep Bayesian Model Discovery ](https://docs.sciml.ai/Overview/stable/showcase/bayesian_neural_ode/)on the Lotka-Volterra model discussed in the [Automated Discovery of Missing Physics example](https://docs.sciml.ai/Overview/stable/showcase/missing_physics/).  The problem I am facing is that I am not able to figure out a way to pass the parameters of the neural network embedded in the ODE of the Lotka-Volterra model to the Hamiltonian as done [here](https://github.com/SciML/SciMLDocs/blob/8660ddc193fea9777b8aa01d5bdf24c2d53b5004/docs/src/showcase/bayesian_neural_ode.md?plain=1#L101).  The main issue here is that the hamiltonian is [fed a vector of parameters ](https://github.com/TuringLang/AdvancedHMC.jl/blob/eb9b2e0d60ef3dd85768d6e6a9f19de15b8f7130/README.md?plain=1#L205) and they are updated naturally as the optimization is carried out. I am having trouble achieving the same with the missing physics example. 
Any pointers as to  how this can be achieved will be very helpful. Thanks.

**Comment by ChrisRackauckas** [comment]
What have you tried so far? You'd do the exact same thing except change the optimization to the Bayesian fitting routine.

**Comment by gurtajbir** [comment]
Hi Chris. I was trying to extract the parameters of the Flux model as a vector using a function like the following for a model U
`p_model = []
for i in 1:length(U.layers)
    weights = Float64.(Flux.params(U.layers[i])[1])
    for row in weights
        p_model = [p_model; row]
    end
    biases = Float64.(Flux.params(U.layers[i])[2])
    p_model = [p_model; biases]
end
    
p_model = Float64.(p_model)`

This was now in the correct format to be fed into `metric = DiagEuclideanMetric(length(p))` and   `integrator = Leapfrog(find_good_stepsize(h, p))`. But the trouble was with converting back to something that could be used by the UDE dynamics.

**Comment by gurtajbir** [comment]
I had posted the same on [Julia discourse]( https://discourse.julialang.org/t/deep-bayesian-model-discovery-without-using-neuralode-object/102403). I was advised to use  `Flux.destructure(chain)` which now enables me to make predictions in the UDE dynamics by using the new set of parameters 
`function ude_dynamics!(du, u, p, t, p_true)
    U = re(p)   # Reconstruct with passed parameters p
    û = U(u)    # Network prediction
    du[1] = p_true[1] * u[1] + û[1]
    du[2] = -p_true[4] * u[2] + û[2]
end`

**Comment by gurtajbir** [comment]
> What have you tried so far? You'd do the exact same thing except change the optimization to the Bayesian fitting routine.

What change are you referring to here? Do you mean a change to the loss `l(θ) = -sum(abs2, ode_data .- predict_neuralode(θ)) - sum(θ .* θ)`

**Comment by ChrisRackauckas** [comment]
I highly recommend just changing to a Lux network instead of a Flux one. Is the tutorial using Flux? If so we should just update it

**Comment by Vaibhavdixit02** [comment]
@gurtajbir it looks like you are on the right track, how many samples is that plot from? Are you dropping the warmup samples?

**Comment by gurtajbir** [comment]
> @gurtajbir it looks like you are on the right track, how many samples is that plot from? Are you dropping the warmup samples?

Hi @Vaibhavdixit02 . This plot was using the below 
`samples, stats = sample(h, prop, p, 500, adaptor, 2000; progress = true)`

**Comment by gurtajbir** [comment]
> I highly recommend just changing to a Lux network instead of a Flux one. Is the tutorial using Flux? If so we should just update it

Hi @ChrisRackauckas. Just changing to Lux seemed to considerably improve how the plot looked. Also, with same network size and activation function, the code with Lux faster than that with Flux (almost 3 times as fast). 
![Screenshot 2023-08-10 at 3 49 32 PM](https://github.com/SciML/SciMLDocs/assets/93142237/ed7d9b05-ac26-46de-9dcc-abd0c0c07bab)

**Comment by ChrisRackauckas** [comment]
Yeah that's expected. Could you make a PR to update that code?

**Comment by gurtajbir** [comment]
Would you like me to update the existing deep bayesian discovery example in the PR or create a code file that resembles what I am trying out on Lotka-Volterra ?

**Comment by ChrisRackauckas** [comment]
Updating the deep bayesian discovery example in a PR that changes it to Lux (and ultimately improves its stability) would be perfect.

**Comment by gurtajbir** [comment]
Sounds good. I'll get started on it.

**Comment by Doublonmousse** [comment]
I've made a gist redoing https://docs.sciml.ai/Overview/stable/showcase/bayesian_neural_ode/#Step-5:-Plot-diagnostics with `Lux` and no `NeuralODE` object here : https://gist.github.com/Doublonmousse/64c5226edc8419dc6bf8e0594f2cb89f.

If this good enough ? I'll do a PR to update the example in this repo if it's okay

**Comment by ChrisRackauckas** [comment]
Yeah that looks good, though I don't think you need the Metal import?

**Comment by Doublonmousse** [comment]
Yeah, I'm not using it (it's also imported twice ..). 

Maybe `vector_to_parameters` is superfluous or can be improved as well? 

I'll open a PR for it then soon so that we can discuss further things there.

**Comment by ChrisRackauckas** [comment]
There is currently an issue with the doc building due to a bug in CUDA https://github.com/SciML/SciMLDocs/pull/224
